imageprocessor/views.py

```diff
@@ -27,8 +29,9 @@ def classify(request):
 			image = Image.open(image_file)
 			context['tags'] = detect(image)
 			return render(request, 'output.html', context)
-		except OSError as err:
-			context['form'] = form
-			return render(request, 'input.html', context)
+		except ValueError:
+			context['errors'] = True
+		except OSError:
```

Comment:
is this unreachable?

Reply:
Nah, they're different types. The ValueError is raised when we can't produce any tags for the image submitted, the OSError is raised when the wrong kind of file is uploaded. I think distinguishing between the two error types should be useful once we've built out our API so that consumers can understand why their request failed. I also plan to iterate on this and potentially use messages, since hardcoding isn't ideal.

Reply:
sorry, for some reason I didnt realize that the other except oserror was being deleted.